### ISSUE TITLE
[Bug] Invalid IDs cause the API to return 500 internal server error

### ISSUE THREAD

**Issue by chungthuang** [dev-launchers/strapiv4#91]
**Describe the bug**
Sentry caught an exception that requests to https://apiv4-staging.devlaunchers.org/api/idea-cards/undefined triggered an exception
```
select "t0".*, "t0"."id", "t0"."created_by_id", "t0"."updated_by_id" from "public"."idea_cards" as "t0" where ("t0"."id" = $1) limit $2 - invalid input syntax for type integer: "undefined"
```

**To Reproduce**
Send a request to https://apiv4-staging.devlaunchers.org/api/idea-cards/undefined 

**Expected behavior**
The API returns 400 bad request.

**Comment by becevka** [comment]
@tiagopazhs Please explore how we can add validation for the requests within Strapi.

**Comment by tiagopazhs** [comment]
I got an insteresting things to explore.

### 1) The first thing is because it is not happening just in `/idea-cards`.
- It is happening in ather routes too like  `/profile`.
- Follow the examplo of profile in 2).

### 2) My enviroments are not reflecting the same answer.
-  In stage i am gotting a 500:
![image](https://github.com/dev-launchers/strapiv4/assets/81580990/09b6754c-374f-4721-a2ec-61744676f927)
- And local I am gotting 404:
![image](https://github.com/dev-launchers/strapiv4/assets/81580990/9d766d25-0107-4e77-9c10-cdb896a44a1d)

**Comment by chungthuang** [comment]
Could the 500 be due to an authentication error? Current we have these permissions for authenticated users in staging for the profile collection:
![image](https://github.com/dev-launchers/strapiv4/assets/7826979/9e941fcd-c809-42cd-9489-543f92e537eb)
What permissions do authenticated users have locally?

**Comment by tiagopazhs** [comment]
Hii, @zar92,

Actually, this issue is not closed.

I am exploring that.

**Comment by tiagopazhs** [comment]
@chungthuang sorry for my late answer. 

I was trying to reproduce authenticated user without success, yeah and we should do it later.

But for now, let's explore just the **Public access** ?:

- Before all, I have a first question, is this the expected access to **PROFILE**? Is there a reason to allow `update` or can we change this?

![image](https://github.com/dev-launchers/strapiv4/assets/81580990/33ccfddb-9010-4e42-9510-f6dabf26aaf3)


- The second question exploring the issue reproduction for itself: I am still trying to access the same route and geting different responses locally `404` and in staging `500`. 
So, do you have another idea about to reproduce that? Remember for now I am testing just the public access. 
Is it a good idea to check the staging logs?

**Comment by zar92** [comment]
@tiagopazhs, I have reopened it.

**Comment by chungthuang** [comment]
@tiagopazhs that's a great observation. Unauthenticated users shouldn't be able to update profiles.
As for the 500 error, were you sending requests to https://apiv4-staging.devlaunchers.org/api/profiles/undefined? I checked sentry and found this error:
```
select "t0".* from "public"."profiles" as "t0" where ("t0"."id" = $1) limit $2 - invalid input syntax for type integer: "undefined"
```

**Comment by becevka** [comment]
@tiagopazhs @chungthuang let's park this issue into Icebox, for now. After we have tests integrated, I think we should utilize testcontainers to emulate staging/production database. Otherwise, we will keep spending precious time back and forth chasing differences in environments.

**Comment by tiagopazhs** [comment]
> 🎉 This issue has been resolved in version 1.7.0 🎉
> 
> The release is available on [GitHub release](https://github.com/dev-launchers/strapiv4/releases/tag/v1.7.0)
> 
> Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot 📦🚀

Good!
